ROCm release 7.0 RC3 (#815)
AMDResearch/omniperf: CHANGELOG.md

Full documentation for ROCm Compute Profiler is available at [https://rocm.docs.amd.com/projects/rocprofiler-compute/en/latest/](https://rocm.docs.amd.com/projects/rocprofiler-compute/en/latest/).

## ROCm Compute Profiler 3.2.1 for ROCm 7.0.0

### Added

> #### MI 350

Reviewer: #### CDNA4 (MI350/MI355) support

Author: > #### CDNA4 (MI350/MI355) support

@feizheng10  is MI350 series CDNA4 or CDNA3?

Author: I believe its CDNA 3, correct me if i am wrong

Full documentation for ROCm Compute Profiler is available at [https://rocm.docs.amd.com/projects/rocprofiler-compute/en/latest/](https://rocm.docs.amd.com/projects/rocprofiler-compute/en/latest/).

## ROCm Compute Profiler 3.2.1 for ROCm 7.0.0

### Added

> #### CDNA4 (MI350/MI355)

Reviewer 2: ```suggestion
#### CDNA4 (AMD Instinct MI350/MI355) support
```

* Deprecation warning for MongoDB database update mode.

* Deprecation warning for ``rocm-smi``

* ``--specs-correction`` option to provide missing system specifications for analysis.

### Changed

* Change the default rocprof version to rocprofv3, this is used when environment variable "ROCPROF" is not set
* Changed ``normal_unit`` default to ``per_kernel``.
* Decreased profiling time by not collecting unused counters in post-analysis.
* Updated Dash to >=3.0.0 (for web UI).
* Changed the condition when Roofline PDFs are generated during general profiling and ``--roof-only`` profiling (skip only when ``--no-roof`` option is present).
* Updated Roofline binaries:
  * Rebuild using latest ROCm stack
  * Minimum OS distribution support minimum for roofline feature is now Ubuntu 22.04, RHEL 9, and SLES15 SP6.

### Optimized

* ROCm Compute Profiler CLI has been improved to better display the GPU architecture analytics

### Resolved issues

* Fixed kernel name and kernel dispatch filtering when using ``rocprofv3``.
* Fixed an issue of TCC channel counters collection in ``rocprofv3``.
* Fixed peak FLOPS of F8, I8, F16, and BF16 on AMD Instinct MI 300.

### Known issues

> 

Reviewer 2: Update to known issues
* On AMD Instinct MI100, accumulation counters are not collected, resulting in the following metrics failing to show up in the analysis: Instruction Fetch Latency, Wavefront Occupancy, LDS Latency
  * As a workaround, use the environment variable ``ROCPROF=rocprof`` , to use ``rocprofv1`` for profiling on AMD Instinct MI100.
* GPU id filtering is not supported when using ``rocprof v3``.
* Analysis of previously collected workload data will not work due to ``sysinfo.csv`` schema change.
  * As a workaround, re-run the profiling operation for the workload and interrupt the process after 10 seconds.
    Followed by copying the ``sysinfo.csv`` file from the new data folder to the old one.
    This assumes your system specification hasn't changed since the creation of the previous workload data.
* Analysis of new workloads might require providing shader/memory clock speed using
``--specs-correction`` operation if ``amd-smi`` or ``rocminfo`` does not provide clock speeds.
* Memory chart on ROCm Compute Profiler CLI might look corrupted if the CLI width is too narrow.

  * L2 to EA stalls
  * L2 to EA stalls per channel

> * Roofline support for MI350 series architecture

Reviewer 2: ```suggestion
* Roofline support for AMD Instinct MI350 series architecture.
```

#### Textual User Interface (TUI) (beta version)

> * Add rocprof-compute Text User Interface (TUI) support for analyze mode

Reviewer 2: ```suggestion
* Text User Interface (TUI) support for analyze mode
```

  * A command line based user interface to support interactive single-run analysis

>   * launch with `--tui` option in analyze mode. i.e., `rocprof-compute analyze --tui`

Reviewer 2: ```suggestion
  * To launch, use `--tui` option in analyze mode. For example, ``rocprof-compute analyze --tui``.
```

#### PC Sampling (beta version)

* Stochastic (hardware-based) PC sampling has been enabled for AMD Instinct MI300X series and later accelerators.

* Host-trap PC Sampling has been enabled for AMD Instinct MI200 series and later accelerators.

> * Support sorting of PC sampling by type: offset or count.

Reviewer 2: ```suggestion
* Support for sorting of PC sampling by type: offset or count.
```

> * PC Sampling Support on CLI and TUI

Reviewer 2: ```suggestion
* PC Sampling Support on CLI and TUI analysis.
```

#### Roofline

> * Support Roofline plot on CLI (single run)

Reviewer 2: ```suggestion
* Support for Roofline plot on CLI (single run) analysis.
```

> * Roofline support for RHEL 10

Reviewer 2: ```suggestion
* Roofline support for RHEL 10 OS.
```

> * Additional data types for roofline profiling

Reviewer 2: Can be removed from ROCm 7.0.0. These are part of ROCm 6.4.2.

Author: Mention that FP4 and FP6 data types have been added to roofline data types for AMD Instinct MI 350 series.

  * Now supports FP4, FP6, FP8, FP16, BF16, FP32, FP64, I8, I32, I64 (dependent on gpu architecture)

#### rocprofv3 support

> * rocprofv3 is the default backend for profiling

Reviewer 2: ```suggestion
* ``rocprofv3`` is supported as the default backend for profiling.
```

> * Add support to obtain performance information for all channels for TCC counters

Reviewer 2: ```suggestion
* Support to obtain performance information for all channels for TCC counters.
```

> * Add support for profiling on MI 100 with rocprofv3

Reviewer 2: ```suggestion
* Support for profiling on AMD Instinct MI100 using ``rocprofv3``.
```

> * Add deprecation warning for rocprofv3 interface in favor of rocprofiler-sdk interface which directly accesses rocprofv3 C++ tool

Reviewer 2: This deprecation should go under Upcoming changes title in the changelog. Eg: ``rocprofv3`` interface is deprecated in favor of the ROCprofiler-SDK interface, which directly accesses ``rocprofv3`` C++ tool. This deprecation warning is also added to the application interface.
```suggestion
* Deprecation warning for ``rocprofv3`` interface in favor of the ROCprofiler-SDK interface, which directly accesses ``rocprofv3`` C++ tool.
```

#### Others

> * Add Docker files to package the application and dependencies into a single portable and executable standalone binary file

Reviewer 2: ```suggestion
* Docker files to package the application and dependencies into a single portable and executable standalone binary file.
```

* Analysis report based filtering

>   * -b option in profile mode now additionally accepts metric id(s) for analysis report based filtering

Reviewer 2: ```suggestion
  * ``-b`` option in profile mode now also accepts metric id(s) for analysis report based filtering.
```

>   * -b option in profile mode also accept hardware IP block for filtering, however, this support will be deprecated soon

Reviewer 2: ```suggestion
  * ``-b`` option in profile mode also accepts hardware IP block for filtering; however, this filter will be deprecated soon.
```

>   * --list-metrics option added in profile mode to list possible metric id(s), similar to analyze mode

Reviewer 2: ```suggestion
  * ``--list-metrics`` option added in profile mode to list possible metric id(s), similar to analyze mode.
```

> * Interface to rocprofiler-sdk

Reviewer 2: ```suggestion
* Interface to ROCprofiler-SDK.
```

>   * Setting ROCPROF=rocprofiler-sdk environment variable will use rocprofiler-sdk C++ library instead of rocprofv3 python script

Reviewer 2: ```suggestion
  * Setting the environment variable ``ROCPROF=rocprofiler-sdk`` will use ROCprofiler-SDK C++ library instead of ``rocprofv3`` python script.
```

* Support MEM chart on CLI (single run)

> * Add deprecation warning for MongoDB database update mode.

Reviewer 2: ```suggestion
* Deprecation warning for MongoDB database update mode.
```

Reviewer 2: Should the deprecation also be part of Upcoming changes?

> * Add deprecation warning for rocm-smi

Reviewer 2: Should this also be part of Upcoming changes?
```suggestion
* Deprecation warning for ``rocm-smi``.
```

> * Add --specs-correction option to provide missing system specification for analysis

Reviewer 2: ```suggestion
* ``--specs-correction`` option to provide missing system specifications for analysis.
```

### Changed

> * Change the default rocprof version to rocprofv3, this is used when environment variable "ROCPROF" is not set

Reviewer 2: ```suggestion
* Changed the default ``rocprof`` version to ``rocprofv3``. This is used when environment variable ``ROCPROF`` is not set.
```

> * Change normal_unit default to per_kernel

Reviewer 2: ```suggestion
* Changed ``normal_unit`` default to ``per_kernel``.
```

> * Decrease profiling time by not collecting counters not used in post analysis

Reviewer 2: This should go under "Optimized" title. 
```suggestion
* Decreased profiling time by not collecting unused counters in post-analysis.
```

> * Update Dash to >=3.0.0 (for web UI)

Reviewer 2: ```suggestion
* Updated Dash to >=3.0.0 (for web UI).
```

> * Change when Roofline PDFs are generated during general profiling and --roof-only profiling (skip only when --no-roof option is present)

Reviewer 2: ```suggestion
* Changed the condition when Roofline PDFs are generated during general profiling and ``--roof-only`` profiling (skip only when ``--no-roof`` option is present).
```

> * Update Roofline binaries

Reviewer 2: ```suggestion
* Updated Roofline binaries:
```

  * Rebuild using latest ROCm stack

>   * OS distribution support minimum for roofline feature is now Ubuntu22.04, RHEL9, and SLES15SP6

Reviewer 2: ```suggestion
  * Minimum OS distribution support for roofline feature is now Ubuntu 22.04, RHEL 9, and SLES15 SP6.
```

### Optimized

* ROCm Compute Profiler CLI has been improved to better display the GPU architecture analytics

### Resolved issues

> * Fixed kernel name and kernel dispatch filtering when using rocprofv3

Reviewer 2: ```suggestion
* Fixed kernel name and kernel dispatch filtering when using ``rocprofv3``.
```

> * Fixed not collecting TCC channel counters in rocprof v3

Reviewer 2: ```suggestion
* Fixed an issue of TCC channel counters collection in ``rocprof v3``.
```

> * Fixed peak FLOPS of F8 I8 F16 and BF16 on MI300

Reviewer 2: ```suggestion
* Fixed peak FLOPS of F8, I8, F16, and BF16 on AMD Instinct MI300.
```

### Removed

* Roofline support for Ubuntu 20.04 and SLES below 15.6

> * Remove support for MI50/MI60 in accordance with the documentation

Reviewer 2: Can we make it clearer by mentioning the documentation in question?
```suggestion
* Removed support for AMD Instinct MI50 and MI60 in accordance with the documentation.
```

> * Removed support for AMD Instinct MI50 and MI60 in accordance with the documentation.

Author: Remove in accordance with the documentation
